costmap_2d/include/costmap_2d/inflation_layer.h

```diff
@@ -103,16 +103,17 @@ class InflationLayer : public Layer
    * @return A cost value for the distance */
   virtual inline unsigned char computeCost(double distance) const
   {
+    double inscribed_radius = inflate_inscribed_ ? inscribed_radius_ : 0.0;
     unsigned char cost = 0;
     if (distance == 0)
       cost = LETHAL_OBSTACLE;
-    else if (distance * resolution_ <= inscribed_radius_)
+    else if (inscribed_radius && distance * resolution_ <= inscribed_radius)
```

Comment:
F:
```suggestion
    else if (inflate_inscribed_ && distance * resolution_ <= inscribed_radius)
```

Reply:
what's the rationale?
to me my version is cleaner, as you check the validity of a value (`inscribed_radius`) right before using it

Reply:
I don't think your way is cleaner. You are implicitly casting a `double` to `bool`, and then checking the `bool` value. That is not checking validity of a value. Your value will always be valid (valid in a sense that it will never be a nullptr or UB).

The way I wrote is more natural imo. Since you are checking a the parameter that is already a boolean (no casting), and basically asking "is it enable?", which is more natural and easy to read.

Reply:
try to read both options as NL:

"if we have inscribed radius and the distance is smaller than it..."

"if we inflate inscribed space and the distance is smaller than the inscribed radius..."

let's someone else to break the tie

Reply:
I don't think you need the check, though. 
`else if (distance * resolution_ <= inscribed_radius)` should be enough. 
Since `distance` should always be `>=0`, `if (inscribed_radius == 0)` you will never enter the `else if`

Reply:
right!